src/EditorFeatures/Core/SymbolSearch/RoslynSymbolResult.cs

```diff
@@ -1,22 +1,73 @@
-﻿using Microsoft.CodeAnalysis.FindUsages;
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
 using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Adornments;
 
 namespace Microsoft.CodeAnalysis.Editor.SymbolSearch
 {
-    internal class RoslynSymbolResult : SymbolSearchResult, IResultInLocalFile
+    internal class RoslynSymbolResult : SymbolSearchResult, IResultInLocalFile, IResultHasClassifiedDefinition, IResultHasClassifiedContext //, IResultOpensWithCustomAction
     {
         private DefinitionItem definition;
+        private SourceReferenceItem reference;
+        private RoslynSymbolSource source;
+        private static ISymbolOriginDefinition LocalSymbolOrigin = null;
```

Comment:
1. `_localSymbolOrigin`.
2. what does 'local' mean in this context?
3. doc these.